php/php.blade/src/org/netbeans/modules/php/blade/project/BladeProjectProperties.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.blade.project;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.prefs.PreferenceChangeListener;
+import java.util.prefs.Preferences;
+import javax.swing.DefaultListModel;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectUtils;
+import org.netbeans.modules.php.blade.editor.ui.customizer.UiOptionsUtils;
+//import org.netbeans.modules.php.blade.editor.actions.ToggleBlockCommentAction;
+import org.openide.util.NbPreferences;
+
+/**
+ * @todo ADD NEW OPTION VALUES
+ * @todo use nodes for files
+ *
+ * @author Haidu Bogdan
+ */
+public final class BladeProjectProperties {
+
+    private static final Map<Project, BladeProjectProperties> INSTANCES = new HashMap<>();
+    private static final String BLADE_VERSION = "blade.version"; // NOI18N
+    private static final String DIRECTIVE_CUSTOMIZER_PATH_LIST = "directive_customizer.path.list"; // NOI18N
+    private static final String VIEW_PATH_LIST = "views.path.list"; // NOI18N
+    private static final String NON_LARAVEL_DECL_FINDER = "non_laravel.decl.finder"; // NOI18N
+    public Project project;
```

Comment:
Why public? This is not common, definitely.

Reply:
Moreover, not thread safe.

Reply:
It's not clear for me.
It's not thread safe, because of the reference to a Project instance ?

Reply:
This class is used from more threads, so you cannot have a field like this. You need to do the threading properly. Please, try to find some information about it, for the IDE development, it is crucial to understand it.

The easiest way how to make a thread-safe class is to define all its fields as `final` ones.